test/nodes/test_multi_node_weighted_sampler.py

```diff
@@ -0,0 +1,203 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import itertools
+from enum import unique
+
+from parameterized import parameterized
+from torch.testing._internal.common_utils import TestCase
+from torchdata.nodes.adapters import IterableWrapper
+
+from torchdata.nodes.samplers.multi_node_weighted_sampler import MultiNodeWeightedSampler
+from torchdata.nodes.samplers.utils import StopCriteria
+
+from .utils import DummyIterableDataset, run_test_save_load_state
+
+
+class TestMultiNodeWeightedSampler(TestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self._num_samples = 10
+        self._num_datasets = 4
+        self._weights_fn = lambda i: 0.1 * (i + 1)
+        self._num_epochs = 3
+
+        self.datasets = {
+            f"ds{i}": IterableWrapper(DummyIterableDataset(self._num_samples, f"ds{i}"))
+            for i in range(self._num_datasets)
+        }
+        self.weights = {f"ds{i}": self._weights_fn(i) for i in range(self._num_datasets)}
+
+    def _setUpMultiNodeWeightedSampler(
+        self, num_samples, num_datasets, weights_fn, stop_criteria
+    ) -> MultiNodeWeightedSampler:
+
+        datasets = {f"ds{i}": IterableWrapper(DummyIterableDataset(num_samples, f"ds{i}")) for i in range(num_datasets)}
+        weights = {f"ds{i}": weights_fn(i) for i in range(num_datasets)}
+        return MultiNodeWeightedSampler(datasets, weights, stop_criteria)
```

Comment:
Let's wrap this in a Prefetcher to make sure it works properly with background threads
```suggestion
        node = MultiNodeWeightedSampler(datasets, weights, stop_criteria)
        return Prefetch(node, prefetch_factor=3)
```